### DISCUSSION TITLE
Potential unintended theme design

### DISCUSSION THREAD

**Discussion by DannyBen** [retypeapp/retype#98] [Q&A]
I am not sure if this is on purpose or just a forgotten CSS, but it seems unintentional that the label in the header is the only element that stays with a light background.

![image](https://user-images.githubusercontent.com/2405099/126826479-67f2ef7a-32d8-4681-a9cd-cdc278e43c39.png)

**Comment by fabriciomurta** [comment]
Well, it actually looked pretty nice to me to both interfaces. So it shouldn't be a styling leftover.

So as long as everywhere else follows this pattern this is not a leftover or unintended theme design.

But again, if I visit https://retype.com/roadmap/ I get
![image](https://user-images.githubusercontent.com/10730372/126844433-5ef4bc78-c3ab-4cc1-9b9b-4fe3c8ec5c09.png)

So it looks like you really have a point there. The styling should logically have followed the `badge` component and although it does not look totally unacceptable to my eyes, it may have been left.

Although I'm not sure the label up there would look as cool if its styling matched the `badge` component. Your mileage may vary.

**Reply by DannyBen** [reply]
I agree it does not look *completely* alien, but the badge color is what I expected to see there, and I am pretty sure it will look great and chocolatey.

**Reply by geoffreymcgill** [reply]
I believe by default the branding `label` uses the `primary` variant colors of the [Badge](https://retype.com/components/badge/) component, and will be the same color in both `light` and `dark` modes. 

The `NEXT` and `LATEST` badges in the screen capture above use the `info` Badge variant, but again, the `info` style and palette does not change between `light` and `dark` modes. 

The `label` colors are also [customizable](https://retype.com/configuration/project/#colors) if you want to adjust:

```json
{
  "branding": {
    "colors": {
      "label": {
        "text": "#ffffff",
        "background": "#ff0000"
      }
    }
  }
}
```

Those same color configs are used in both `light` and `dark` modes, but we're doing some work with custom color palettes and other UI customizations in the coming releases, so I will make a point of adding separate `light` and `dark` mode configs for the label colors.

Hope this helps.

**Reply by DannyBen** [reply]
> I believe by default the branding label uses the primary variant colors of the Badge component

Oh boy... then the problem is worse in my mind. This is very out of character for a decent dark theme. In my mind, color relationships should remain the same between dark and light (i.e. mirrored) - with one exception: If the light theme uses dark code blocks, then the dark theme continues to use dark code blocks.

But - for things like badhes, I expect this relationship:

Since the light theme has the badges background at a very slightly darker color, I expect the dark them to be the exact mirror and have the badges background color slightly lighter than the body color.

As you mentioned color config impacts both themes, so it renders it less useful and it ruins the smooth theme experience.

**Reply by geoffreymcgill** [reply]
meh. I don't necessarily agree. There's nothing saying the colors HAVE to change between `light` and `dark` modes. Some color combos work great across both. It just so happens that the default Badge color variant (`primary`) works great in both `light` and `dark`.

> Since the light theme has the badges background at a very slightly darker color, I expect the dark them to be the exact mirror and have the badges background color slightly lighter than the body color.

Five out of the nine [variants](https://retype.com/components/badge/#variants) do adjust between `light` and `dark` modes, such as `secondary`, `info`, `light`, `dark`, and `contrast`. The other four work great across both themes.

Of course we could make the `primary` variant slightly darker in `light` mode and slightly lighter in `dark` mode, but why? It would just add more css weight to the page and isn't required.

That said, the variant palettes are certainly something we can review again.
